Sources/TensorFlow/Operators/Comparison.swift

```diff
@@ -227,3 +227,16 @@ public extension StringTensor {
         return Raw.equal(self, other)
     }
 }
+
+public extension Tensor where Scalar: FloatingPoint {
+    /// Returns `true` if tensors are of equal shape and all pairs of scalars are approximately
+    /// equal.
+    @inlinable
+    func isAlmostEqual(
+        to other: Tensor,
+        tolerance: Scalar = Scalar.ulpOfOne.squareRoot()
+    ) -> Bool {
+        return self.shape == other.shape &&
+            zip(self.scalars, other.scalars).allSatisfy { abs($0.0 - $0.1) < tolerance }
```

Comment:
Note: this implementation is a bit simpler than the [floating-point approximate equality pitch](https://forums.swift.org/t/approximate-equality-for-floating-point/22420) implementation, which performs two kinds of scaling (one to handle infinite values, one in the general case).

If the pitch becomes an accepted proposal, we may want to adopt its implementation.

Reply:
Do you want me to adopt the same scaling? Should we add this to `elementsApproximatelyEqual` and then call that here as suggested by @eaplatanios ?

Reply:
For simplicity, not adopting for now SGTM.
We can re-evaluate if [SE-0259](https://github.com/apple/swift-evolution/blob/master/proposals/0259-approximately-equal.md) is accepted.